scripts/did_finder.py

```diff
@@ -26,157 +26,91 @@
-    handler = logging.StreamHandler()
-    handler.setFormatter(formatter)
-    handler.setLevel(logging.INFO)
-    log.addHandler(handler)
-    log.setLevel(logging.INFO)
-    return log
-
-
-# RabbitMQ Queue Callback Method
-def callback(channel, method, properties, body):
-    try:
-        did_request = json.loads(body)
-        print("----> ", did_request)
-        servicex = ServiceXAdapter(did_request['service-endpoint'])
-        did = did_request['did']
-        request_id = did_request['request_id']
-
-        servicex.post_status_update("DID Request received")
-        lookup_request = LookupRequest(
-            request_id=request_id,
-            did=did,
-            rucio_adapter=rucio_adapter,
-            servicex_adapter=servicex,
-            site=site,
-            prefix=prefix,
-            chunk_size=1000,
-            threads=threads
-        )
-
-        lookup_request.lookup_files()
-
-    except Exception:
-        traceback.print_exc()
-        exc_type, exc_value, exc_traceback = sys.exc_info()
-        servicex.post_status_update("DID Request failed " + str(exc_value))
-    finally:
-        channel.basic_ack(delivery_tag=method.delivery_tag)
-
-
-def init_rabbit_mq(rabbitmq_url, retries, retry_interval):
-    rabbitmq = None
-    retry_count = 0
-
-    while not rabbitmq:
-        try:
-            rabbitmq = pika.BlockingConnection(pika.URLParameters(rabbitmq_url))
-            _channel = rabbitmq.channel()
-            _channel.queue_declare(queue=QUEUE_NAME)
-
-            print("Connected to RabbitMQ. Ready to start consuming requests")
-
-            _channel.basic_consume(queue=QUEUE_NAME,
-                                   auto_ack=False,
-                                   on_message_callback=callback)
-            _channel.start_consuming()
+from servicex_did_finder_lib import default_command_line_args, start_did_finder
+from servicex.did_finder.lookup_request import LookupRequest
 
-        except pika.exceptions.AMQPConnectionError as eek:
-            rabbitmq = None
-            retry_count += 1
-            if retry_count < retries:
-                print("Failed to connect to RabbitMQ. Waiting before trying again")
-                time.sleep(retry_interval)
-            else:
-                print("Failed to connect to RabbitMQ. Giving Up")
-                raise eek
 
+async def find_files(rucio_adaptor: RucioAdapter, site: str, prefix: str, threads: int,
+                     did_name: str, info: Dict[str, Any]) \
+                     -> AsyncGenerator[Dict[str, Any], None]:
+    '''For each incoming did name, generate a list of files that ServiceX can process
+    from the rucio catalog
+
+    Args:
+        did_name (str): Rucio Dataset name
+        info (Dict[str, Any]): Various bits of info about the request.
+
+    Returns:
+        AsyncGenerator[Dict[str, any], None]: yield each file
+    '''
+
+    lookup_request = LookupRequest(
```

Comment:
Can this be moved into `run_rucio_finder`?

Reply:
Yes! I'll move this in the next check-in.